### PR TITLE
Semi-coherent zdc benchmark

### PR DESCRIPTION
**Pull request by JiaJunHuang120000**
### Briefly, what does this PR introduce?
Semi-coherent events analysis for the zdc lyso.

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [x] Other: _Added one plot for the energy spectrum of photons in the zdc lyso_

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [x] Changes have been communicated to collaborators @bschmookler 

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No.
### Does this PR change default behavior?
No.

### CODE REVIEW

File: benchmarks/semi_coherent/Snakefile [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import os
+
+rule semi_coherent_sim:
+	input:
+		hepmcfile="benchmarks/semi_coherent/input.hepmc",
```
Comment (veprbl) [inline benchmarks/semi_coherent/Snakefile:5]: Could you provide the script that does the filtering starting from some input that is available on S3?

We had used some hardcoded input files before, in https://eicweb.phy.anl.gov/EIC/datasets , but they never get updated and are a bit of a black box.

Comment (JiaJunHuang120000) [reply benchmarks/semi_coherent/Snakefile:5]: Hi,

    In the following attachments I included the python scipt that I used for filtering the semi-coherent events from the original BeAGLE file after hepmc transformation, and a modified Snakefile that uses S3 data and applying the filtering script with afterburner.
    
[semi_coherent.zip](https://github.com/user-attachments/files/16515772/semi_coherent.zip)

    However I did encounter a problem trying to apply afterburner to the filtered `ePb_beagle.hepmc` file that was in `eictest/EPIC/EVGEN/CI`, and we think that the file might be too old. We would like to know if there is any other newer BeAGLE files on S3 with afterburner applied. If not can we upload our own files to S3.

Comment (veprbl) [reply benchmarks/semi_coherent/Snakefile:5]: I've looked at the semi_coherent/Snakefile.py from the zip file. You are manually modifying text format for HepMC (v3?). That will not work with files on S3, as they are in the ROOT HepMC format. Could you rewrite that script using the HepMC bindings?

Here is an example snippet on how to read a ROOT HepMC event in python:
https://github.com/eic/detector_benchmarks/blob/a40a48bbfc0e0db6932cd22d96b5219cfc7879c2/benchmarks/backgrounds/ecal_backwards.org?plain=1#L92-L97
Writing should be similar (you could write to text format, if you want).

Comment (JiaJunHuang120000) [reply benchmarks/semi_coherent/Snakefile]: Hi Dmitry,

I have just updated this branch to grab and filter the .hepmc file directly from S3. The beagle file in S3 is in hepmc format and not in ROOT hepmc format.

**Comment by bschmookler** [comment]
> @JiaJunHuang120000 have you tested this? I'm getting an error in abconv, also it's on CI: https://eicweb.phy.anl.gov/EIC/benchmarks/physics_benchmarks/-/jobs/3913962

The problem seems to be due to the BeAGLE file on S3/xrdroot. It is an older BeAGLE file. If we generate a BeAGLE file ourselves locally, convert it to HepMC format, and then run abconv, everything works fine.

The only eA BeAGLE file we can find on S3/xrdroot is this one:

```
xrdfs root://dtn-eic.jlab.org ls /work/eic2/EPIC/EVGEN/CI/ePb_beagle.hepmc
/work/eic2/EPIC/EVGEN/CI/ePb_beagle.hepmc
```

Are there any other ones available? Or can we have write permission to upload some BeAGLE files that we generated?

**Comment by veprbl** [comment]
@bschmookler Wild idea: I assume we have beagle installed in the container. Could you script hepmc3 generation into the benchmark?

**Comment by bschmookler** [comment]
I don't see it in `/opt/local/bin/`. Would it be somewhere else in the container?

**Comment by veprbl** [comment]
Okay, I now remember. We can't get beagle in the container if it depends on fluka. Can you share your file then? I can get it onto XRootD myself.

**Comment by JiaJunHuang120000** [comment]
Hi,

    I have updated the snakefile to take files from JLab XRootD, but there is some changes to the \master branch which I am not sure how to account for. Please let me know what is different. Thanks!

File: benchmarks/Exclusive-Diffraction-Tagging/semi_coherent/filter.py
Comment (veprbl) [inline benchmarks/Exclusive-Diffraction-Tagging/semi_coherent/filter.py:1]: Using textual representation of HepMC3 is not a good idea. Please, re-write this using
```python
import HepMC3
```

Comment (JiaJunHuang120000) [reply benchmarks/Exclusive-Diffraction-Tagging/semi_coherent/filter.py:1]: Is the module install inside ./eic-shell? I receive an error while importing `ModuleNotFoundError: No module named 'HepMC3'`.

Comment (veprbl) [reply benchmarks/Exclusive-Diffraction-Tagging/semi_coherent/filter.py:1]: Oh, looks like it's `import pyHepMC3`

Comment (JiaJunHuang120000) [reply benchmarks/Exclusive-Diffraction-Tagging/semi_coherent/filter.py:1]: Hi,

    Can we merge this branch first now that and changes have been settle, and later on we can modify the for using the module?

Comment (veprbl) [reply benchmarks/Exclusive-Diffraction-Tagging/semi_coherent/filter.py:1]: Yes, go ahead.

Comment (veprbl) [reply benchmarks/Exclusive-Diffraction-Tagging/semi_coherent/filter.py:1]: @JiaJunHuang120000 Any follow up on this?

**Pull request merged**